projects/baby_driver/scripts/can_util.py

```diff
@@ -0,0 +1,129 @@
+"""This module implements a low-level wrapper over Python-CAN's CAN API specific to Babydriver."""
+
+import time
+import can
+
+from message_defs import BABYDRIVER_DEVICE_ID, BABYDRIVER_CAN_MESSAGE_ID
+
+
+# The default CAN channel to use for this module. Changed dynamically by cli_setup.
+default_channel = "can0" # pylint: disable=invalid-name
+
+
+def get_bus(channel=None):
+    """Returns a new Python-CAN Bus for sending/receiving messages."""
+
+    if channel is None:
+        channel = default_channel
+
+    return can.interface.Bus(bustype="socketcan", channel=channel, bitrate=500000)
+
+
+def get_arbitration_id(msg_id, device_id):
+    """Returns the arbitration ID for a CAN message from its message and device ID."""
+    # our CAN system uses 6 bits of message ID, 1 bit for ACK/DATA, and 4 bits for device ID
+    return (msg_id << 5) | device_id
+
+
+def arbitration_to_message_id(arbitration_id):
+    """Extracts and returns the message ID from a CAN message's arbitration ID."""
+    # message ID is bits 5-11
+    return (arbitration_id >> 5) & 0b111111
+
+
+def send_message(
+        babydriver_id,
+        data,
+        channel=None,
+        msg_id=BABYDRIVER_CAN_MESSAGE_ID,
+        device_id=BABYDRIVER_DEVICE_ID,
+):
+    """Sends a CAN message.
+
+    Args:
+        babydriver_id: The babydriver ID (first byte of message data) of the message to send.
+        data: Up to 7 bytes (0-255) of data to send in the CAN message.
+        channel: The SocketCAN channel on which to send the message.
+        msg_id: The CAN message ID to use.
+        device_id: The device ID to use.
+
+    Raises:
+        can.CanError: If there was an error in transmitting the message.
+    """
+
+    if len(data) > 7:
+        raise ValueError("Only 7 bytes of data may be sent")
+    if len(data) < 7:
+        # pad to 7 bytes so that the firmware project will accept it
+        data += [0] * (7 - len(data))
```

Comment:
should data=None be handled here? I dunno python, does None + list = list? I guess the function could just require passing an empty list, but seems conveniently descriptive to pass None for data